playwright.config.js

```diff
@@ -0,0 +1,103 @@
+// @ts-check
+import { defineConfig, devices } from "@playwright/test";
+
+/**
+ * @see https://playwright.dev/docs/test-configuration
+ */
+export default defineConfig({
+  testDir: "./playwright-tests/tests",
+  /* Maximum time one test can run for. */
+  timeout: 30 * 1000,
+  expect: {
+    /**
+     * Maximum time expect() should wait for the condition to be met.
+     * For example in `await expect(locator).toHaveText();`
+     */
+    timeout: 5000,
+  },
+  /* Run tests in files in parallel */
+  fullyParallel: true,
+  /* Fail the build on CI if you accidentally left test.only in the source code. */
+  forbidOnly: !!process.env.CI,
+  /* Retry on CI only */
+  retries: process.env.CI ? 2 : 0,
+  /* Opt out of parallel tests on CI. */
+  workers: process.env.CI ? "100%" : undefined,
+  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
+  reporter: "line",
+  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
+  use: {
+    video: "off",
+    /* Maximum time each action such as `click()` can take. Defaults to 0 (no limit). */
+    actionTimeout: 0,
+    /* Base URL to use in actions like `await page.goto('/')`. */
+    baseURL: "http://localhost:4040",
+
+    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
+    trace: "on-first-retry",
+    storageState: {
+      cookies: [],
+      origins: [
+        {
+          origin: "http://localhost:4040",
+          localStorage: [
+            {
+              name: "flags",
+              value: JSON.stringify({
+                bosLoaderUrl: "http://127.0.0.1:4040/api/loader",
+              }),
+            },
+          ],
+        },
+      ],
+    },
+  },
+
+  /* Configure projects for major browsers */
+  projects: [
+    //{
+    // name: "chromium",
+    // use: { ...devices["Desktop Chrome"] },
+    // },
+
+    /*{
+      name: 'firefox',
+      use: { ...devices['Desktop Firefox'] },
+    },
+
+    {
+      name: 'webkit',
+      use: { ...devices['Desktop Safari'] },
+    },*/
+
+    /* Test against mobile viewports. */
+    // {
+    //   name: 'Mobile Chrome',
+    //   use: { ...devices['Pixel 5'] },
+    // },
+    // {
+    //   name: 'Mobile Safari',
+    //   use: { ...devices['iPhone 12'] },
+    // },
+
+    /* Test against branded browsers. */
+    // {
+    //   name: 'Microsoft Edge',
+    //   use: { channel: 'msedge' },
+    // },
+    {
+      name: "Google Chrome",
+      use: { channel: "chrome" },
+    },
+  ],
+
+  /* Folder for test artifacts such as screenshots, videos, traces, etc. */
+  outputDir: "test-results/",
+
+  /* Run your local dev server before starting the tests */
+  webServer: {
+    command: "npm run dev",
+    port: 4040,
```

Comment:
I think this port needs to match the dev server, which is 8000. Change all 4040 to 8000

Reply:
Hey! when i changed the dev server to 3000 it worked . Is it correct ?
I went through the documentation of canvas and found this

![image](https://github.com/near-everything/canvas/assets/91129111/71d84279-7217-4014-9ac6-d4a5b2afdc27)